mods/explorer-double-f2-rename-extension.wh.cpp

```diff
@@ -31,24 +32,106 @@ This works great together with [extension-change-no-warning](https://windhawk.ne
 // ChatGPT and the Windhawk Discord server.
 
 #include <string>
-
-DWORD explorerThreadId;
-HHOOK keyboardHook;
+#include <vector>
+
+static void SelectExtension(HWND editControl) {
+    // typical max filename length
+    std::wstring buffer(260, L'\0');
+    int copied = GetWindowTextW(editControl, buffer.data(), (int)buffer.size());
+    if (copied <= 0) {
+        return;
+    }
+    // trim to actual length
+    buffer.resize(copied);
+
+    size_t dotIndex = buffer.find_last_of(L'.');
+    if (dotIndex > 0 && dotIndex != std::wstring::npos) {
```

Comment:
I didn't try the mod yet, but from the code I'd assume that the `dotIndex > 0` condition will make it so that an extension isn't selected for file names starting with periods, e.g. `.git`. If my understanding is correct, I think it's better to remove that check.

Reply:
I was on the fence about it but in the end I decided to keep it this way because the tail part of dotfiles isn't really a filetype extension, it's just its meaningful name and the dot is only there because someone on unix thought it shouldn't show up in `ls`. Files change extension all the time because they're mis-mimetyped, but renaming a dotfile turns it into something totally different in purpose.

On the other hand, purely lexicographically it would be more consistent, and it's not like there's anything else of use that double-f2 could do on a dotfile... I'll change it tomorrow